src/query/numeric_opfunc.c

```diff
@@ -3871,15 +3873,36 @@ numeric_db_value_print (const DB_VALUE * val, char *buf)
       /* Add the negative sign */
       if (temp[i] == '-')
 	{
-	  buf[nbuf] = '-';
-	  nbuf++;
+	  buf[nbuf++] = '-';
 	}
 
       /* Add decimal point */
       if (i == temp_size - scale)
 	{
-	  buf[nbuf] = '.';
-	  nbuf++;
+	  int k = temp_size - 1;
+
+	  if (oracle_style_number)
+	    {
```

Comment:
It would be better to change it like below.

```suggestion
	  if (oracle_style_number)
	    {
	      int k = temp_size - 1;
```